pkg/render/node.go

```diff
@@ -255,137 +289,180 @@ func (c *nodeComponent) nodeRoleBinding() *rbacv1.ClusterRoleBinding {
-				Resources: []string{"pods", "nodes", "namespaces"},
-				Verbs:     []string{"get"},
-			},
-			{
-				// Used to discover Typha endpoints and service IPs for advertisement.
-				APIGroups: []string{""},
-				Resources: []string{"endpoints", "services"},
-				Verbs:     []string{"watch", "list", "get"},
-			},
-			{
-				// Some information is stored on the node status.
-				APIGroups: []string{""},
-				Resources: []string{"nodes/status"},
-				Verbs:     []string{"patch", "update"},
-			},
-			{
-				// For enforcing network policies.
-				APIGroups: []string{"networking.k8s.io"},
-				Resources: []string{"networkpolicies"},
-				Verbs:     []string{"watch", "list"},
-			},
-			{
-				// Metadata from these are used in conjunction with network policy.
-				APIGroups: []string{""},
-				Resources: []string{"pods", "namespaces", "serviceaccounts"},
-				Verbs:     []string{"watch", "list"},
-			},
-			{
-				// Calico patches the allocated IP onto the pod.
-				APIGroups: []string{""},
-				Resources: []string{"pods/status"},
-				Verbs:     []string{"patch"},
-			},
-			{
-				// Calico needs to query configmaps for pool auto-detection on kubeadm.
-				APIGroups: []string{""},
-				Resources: []string{"configmaps"},
-				Verbs:     []string{"get"},
-			},
-			{
-				// For monitoring Calico-specific configuration.
-				APIGroups: []string{"crd.projectcalico.org"},
-				Resources: []string{
-					"bgpconfigurations",
-					"bgppeers",
-					"blockaffinities",
-					"clusterinformations",
-					"felixconfigurations",
-					"globalnetworkpolicies",
-					"stagedglobalnetworkpolicies",
-					"globalnetworksets",
-					"hostendpoints",
-					"ipamblocks",
-					"ippools",
-					"networkpolicies",
-					"stagedkubernetesnetworkpolicies",
-					"stagednetworkpolicies",
-					"networksets",
-				},
-				Verbs: []string{"get", "list", "watch"},
-			},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "calico-vpp-node-role",
+		},
+		Subjects: []rbacv1.Subject{
 			{
-				// For migration code in calico/node startup only. Remove when the migration
-				// code is removed from node.
-				APIGroups: []string{"crd.projectcalico.org"},
-				Resources: []string{
-					"globalbgpconfigs",
-					"globalfelixconfigs",
-				},
-				Verbs: []string{"get", "list", "watch"},
+				Kind:      "ServiceAccount",
+				Name:      "calico-vpp-node-sa",
+				Namespace: common.VppNamespace,
 			},
-			{
-				// Calico creates some configuration on startup.
-				APIGroups: []string{"crd.projectcalico.org"},
-				Resources: []string{
-					"clusterinformations",
-					"felixconfigurations",
-					"ippools",
-				},
-				Verbs: []string{"create", "update"},
+		},
+	}
+	return crb
+}
+
+// nodeRole creates the clusterrole containing policy rules that allow the node daemonset to operate normally.
+func (c *nodeComponent) nodeRole(forVpp bool) *rbacv1.ClusterRole {
+	var role *rbacv1.ClusterRole
+	if !forVpp {
```

Comment:
I'm not a fan of this approach for re-using the ClusterRole struct between two different cluster role. I think we should just have two separate cluster roles.